tests/local/test_ScraperExtractFromTextTest.py

```diff
@@ -0,0 +1,68 @@
+import json
+import os
+import unittest
+from glob import glob
+from typing import List
+
+from juriscraper.lib.importer import build_module_list
+
+
+class ScraperExtractFromText(unittest.TestCase):
+    """Adds specific tests to specific courts that are more-easily tested
+    without a full integration test.
+    """
+
+    def run_tests_on_module_str(self, module_str: str) -> List[str]:
+        """Finds all the $module_example* files and tests them with the sample
+        scraper.
+        """
+        return build_module_list(module_str)
+
+    def test_extract_from_text_properly_implemented(self):
+        """Ensure that extract_from_text is properly implemented."""
+
+        module_strings = self.run_tests_on_module_str("juriscraper.opinions")
+        for module_string in module_strings:
+            if "backscraper" in module_string:
+                continue
+            package, module = module_string.rsplit(".", 1)
+            mod = __import__(
+                f"{package}.{module}", globals(), locals(), [module]
+            )
+            site = mod.Site()
+            try:
+                if site.extract_from_text("This is a test.") == {}:
```

Comment:
Unless I'm mistaken, wouldn't the `extract_from_text` method return an empty dict if, say, it didn't find the text it was looking for — for example, if the fixture was no good? 

Assuming that's the case, would it make sense to actually check if the method is overridden (instead of based on its outputs)? I [think](https://stackoverflow.com/a/44914111/64911) you can do so with something like:

```
if Site.extract_from_text == OpinionSite.extract_from-text:
    # not overridden
    continue
```

Reply:
Oh, and sorry, one other thing, if it's the case that it *is* overridden, I don't think I see how we enforce that there are tests. Could we also assert that the number of test cases is greater than zero if the method is overridden?

Reply:
No.  It wouldn't return {}.  

In the BIA context if it didn't it would return. Either NONE because it failed or return the following dict where date_filed would be incorrect I think.  Either way an empty {} is not what would be returned.  

```
metadata = {
            "OpinionCluster": {
                "date_filed": date_filed,
                "date_filed_is_approximate": False,
            },
        }
```

Reply:
OK, so I'm confused what this try/if/except block is doing, in that case, sorry?

Reply:
If I pass in a random string.  if we haven't implemented it -... its going to return `{}`. so skip.  But if we have implemented it ... its going to crash.